cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -143,6 +146,43 @@ TEST_F(JITLibTests, LogicalOpTest) {
       false, true, true, [](JITValue& a, JITValue& b) { return !a; });
 }
 
+TEST_F(JITLibTests, TupleTest) {
+  LLVMJITModule module("Test Module");
+  JITTuple layer1(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+  JITTuple layer2(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+  JITTuple layer3(JITTypeTag::TUPLE, JITBackendTag::LLVMJIT);
+
+  JITFunctionPointer function1 = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_Tuple",
+      .ret_type = JITFunctionParam{.type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "a", .type = JITTypeTag::INT32}}});
+
+  JITValuePointer x_in = function1->createVariable("x1", JITTypeTag::INT32);
+  JITValuePointer a_in = function1->getArgument(0);
+  JITValuePointer b_in = function1->createConstant(JITTypeTag::INT32, 876);
+
+  layer2.append<JITValue>(b_in.get());
```

Comment:
Hmm, seems could overload a ```append``` method for ```JITValue&``` so that the call of ```get``` could be omitted.

Reply:
will use a ```BaseValue``` reference instead of a template pointer

Reply:
we get a raw point from unique_ptr and save it in `JITTuple`, it's not safe.